### PR TITLE
upcoming: [M3-7294] - Add AGLB Routes Step to Full Create Flow

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Adds **Route** support to the AGLB full create flow
- Updates service targets to be scoped to each configuration based on UX request

> [!note]
> This PR adds **Routes**, but does not support rules yet. I will add rule functionality in a separate PR to keep the size down. Rules are complex because of the drag and drop.

## Preview 📷

Screenshot 2023-12-13 at 12 14 06 PM

### Prerequisites
- Turn on the `AGLB Full Create Flow` feature flag
- Go to `http://localhost:3000/loadbalancers/create`

### Verification steps 
- Verify you can **add, update, and remove** routes on the AGLB create flow

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

**Review by cpathipa — COMMENTED** [review]
LGMT, functionality is working as expected!

- [X] - Verified add, update, and remove routes on the AGLB create flow.
- [X] - Confirmed state is updating as expected. 
image

Noticed console log error while creating service targets followed by routes. 
image

**Review by coliu-akamai — COMMENTED** [review]
this looks awesome! great work Banks 🎉 
✅ verified adding, removing, editing routes

A couple things I noticed, but they might be unrelated to this PR/out of scope:

The search bar 
a) changes size whenever I start typing in it
b) a console error pops up -- changing from uncontrolled to controlled input (you can see in the video it pops up)

| search bar size change  | console error   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/139280159/e208f4be-9803-4136-ae35-f80e0a6ef426" /> | image |

File: packages/manager/src/features/LoadBalancers/LoadBalancerCreate/EditRouteDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const { configurationIndex, onClose: _onClose, open, routeIndex } = props;
+
+  const {
+    setFieldValue,
+    values,
+  } = useFormikContext<LoadBalancerCreateFormData>();
+
+  const allOtherRoutes = values.configurations.reduce<RoutePayload[]>(
+    (acc, configuration, configIndex) => {
+      const otherRoutes = configuration.routes!.filter((r, index) => {
```
[the remark this answers is not in the document]
Comment (coliu-akamai) [reply packages/manager/src/features/LoadBalancers/LoadBalancerCreate/EditRouteDrawer.tsx:29]: this is a very small nitpick so feel free to ignore lol

File: packages/manager/src/features/LoadBalancers/LoadBalancerCreate/LoadBalancerConfigurations.tsx [hunk 2]
```diff
@@ -1,17 +1,111 @@
...
+    setSelectedServiceTargetIndex(undefined);
+    setIsServiceTargetDrawerOpen(false);
+  };
+
+  const handlers: Handlers = {
+    handleAddRoute,
+    handleAddServiceTraget,
+    handleCloseServiceTargetDrawer,
+    handleEditRoute,
+    handleEditServiceTraget,
```
Comment (mjac0bs) [inline packages/manager/src/features/LoadBalancers/LoadBalancerCreate/LoadBalancerConfigurations.tsx:74]:
```suggestion
    handleEditServiceTarget,
```

**Review by coliu-akamai — APPROVED** [review]
thanks for the fixes Banks! approving pending Mariah's comments addressed 🎉

File: packages/manager/src/features/LoadBalancers/LoadBalancerCreate/Routes.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            InputProps={{
+              endAdornment: query && (
+                <InputAdornment position="end">
+                  <IconButton
+                    aria-label="Clear"
+                    onClick={() => setQuery('')}
+                    size="small"
+                    sx={{ padding: 'unset' }}
+                  >
+                    <CloseIcon sx={{ color: '#aaa !important' }} />
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/features/LoadBalancers/LoadBalancerCreate/Routes.tsx:70]: I think it was the only way I could easily achieve what the UX mockups showed. As AGLB progresses, I'll likely turn this "filter"  TextField into a more reusable component. It's currently duplicated a lot 😖

**Pull request merged**